### PR TITLE
Documentation/adding ds and general documentation

### PR DESCRIPTION
**Pull request by mikayla-kosmala**
## Description

Created a bunch of documentation for the next cohort to be able to read to be able to stay up to date with what is going on.

**App_folder_documenation**
- Contains some information about all the files living in the app folder documentation
### General documentation folder:
**Intake form documentation**
- This contains all google docs and dataflow charts about the intake form
- All the tickets that have been worked on to get the intake forms looking better

**DS api contract** 
- This contains our link to elasticbeanstalk since that is currently the source of truth for the DS API and needs to be communicated across all teams.

[Loom Video](https://www.loom.com/share/e2096524a7574ce5b2fbe0415c3db4e8) 

## Type of change

- [x] New documentation

## Checklist:

- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by tammer81 — APPROVED** [review]
A much-needed resource; thank you for putting this together

**Review by VABIII — APPROVED** [review]
Excellent job untangling all that

**Review by Lopez-John — APPROVED** [review]
Great way to document what was done for the next cohort coming through. Amazing work.

**Comment by Ike888** [comment]
This is great; thank you for all your hard work!

**Comment by benedictSchurwanz** [comment]
This is really so cool

**Review by kevinbui100 — APPROVED** [review]
I love what you've done here. You are a trailblazer!

**Review by Ike888 — APPROVED** [review]
Mikayla, you have done a fabulous job; future developers will thank you for this!

**Review by Ike888 — APPROVED** [review]
This is a great start to having all the documentation in just one place.

**Review by GermanParra — APPROVED** [review]
Awesome work! , well organized, clear, and objective. This document would help cohorts get familiar with the code base quickly.

**Review by BobbyWilt — APPROVED** [review]

File: app/readme.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+2. Easily provide correct answers. If it’s not easy to find no one will use it anyway.
+3. Not replace human interaction. Individuals and interaction over processes and tools, right?
+
+## api.py
+
+This api contains all of our current endpoints that connect backend to ds. Backend uses these endpoints to send and receive data from the data science team. For more detail on each endpoint visit our [elasticbeanstalk API URL](http://underdog-devs-ds-a-dev.us-east-1.elasticbeanstalk.com) or open the file and look at doc strings.
+
+## data.py
+
+Not 100% sure what this file contains, but it seems important to working mongodb
```
Comment (jinjahninjah) [inline app/readme.md:14]: `data.py` is where the MongoDB interface (CRUD Operations) are housed.

File: app/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+This file current contains the function that is used in the /graphs/tech-stack-by-role endpoint.
+
+## model.py
+
+Here is where our matching classes are located. There are a few matching features implemented and they are stored here. Here is a list of the current classes that we have and what they do. For more detail on each class please open the file and look at doc strings. 
+
+MatcherSortSearch
+
+MatcherSortSearchResource
+
```
Comment (jinjahninjah) [inline app/readme.md:27]: Lines 24-27 seem to be out of place-- I'm not sure what they are. Remove this as you call out more info in the doc strings.

File: app/readme.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+## utilities.py
+
+This file current contains the function that is used in the /financial_aid/{profile_id} endpoint.
```
Comment (jinjahninjah) [inline app/readme.md:30]: current -> currently

**Review by jinjahninjah — CHANGES_REQUESTED** [review]
Just a couple of edits in the comments then it should be good to go 👍

**Review by jinjahninjah — APPROVED** [review]

**Pull request merged**